test/15-elements-maps.js

```diff
@@ -71,6 +71,16 @@ test('do not encode undefined in a map', function (t) {
   t.end()
 })
 
+test('do not encode NaN in a map', function (t) {
+  var instance = msgpack()
+  var expected = { hello: 'world' }
+  var toEncode = { a: NaN, hello: 'world' }
+  var buf = instance.encode(toEncode)
+
+  t.deepEqual(expected, instance.decode(buf), 'must ignore NaN')
```

Comment:
why? Shouldn't it have the same behavior as encoded directly?

Reply:
did it same as for `undefined` value

Reply:
They are different. The behavior of `undefined`  is matching how JSON.stringify() works. `undefined`  in javascript means "there is nothing here". `NaN`  has a different meaning altogether. It should throw anyway.

Reply:
Ok, will remove that